.github/dependabot.yml

```diff
@@ -18,7 +18,11 @@ updates:
     schedule:
       interval: weekly
     allow:
-      - dependency-type: all
+      - dependency-type: direct
+    groups:
+      tools-mod:
+        patterns:
+          - "*"
```

Comment:
This seems reasonable to me, most of our purely indirect dependencies come from tools so we end up closing them. Thanks for proposing it @mmorel-35, it should reduce weekly spam pr's.   What do you think @ivanvc?

Reply:
I think this PR has some overlap with the work from issue https://github.com/etcd-io/etcd/issues/18925.

That aside, I tested it in my fork, and it seems like dependabot is tidying the module correctly (https://github.com/ivanvc/etcd/pull/450), as there is only a single directory specified (it looks like the tidying bug is only present when there is more than one directory specified).

The only potential risk I see is that if we have a dependency in `/tools/mod` that is also used in other submodule (i.e., `github.com/gogo/protobuf`, `go.etcd.io/raft/v3`, `github.com/grpc-ecosystem/grpc-gateway/v2`), our CI check `pull-etcd-verify` (`verify-dep`), will fail because the dependencies won't be at the same version.

Reply:
An alternative would be first to merge the manual updates (including updating `/tools/mod` dependencies that are used in other modules). Then, wait for dependabot to update the grouped pull request. Finally, we could merge the grouped `/tools/mod` PR.

Even though it can be an alternative, it may not be viable (as just writing the steps in the previous paragraph took me a while to ensure I got it right). This alternative could also be potentially confusing, and instead of decreasing the toil, it could possibly increase it :sweat_smile:... But I'm trying to think of alternatives to make this work.

Reply:
So, I'm in favor of keeping `dependency-type: direct` but dropping the group. What do you think, @jmhbnz?

Reply:
Sounds good to me, let's go with just the `dependency-type: direct` for this pr @mmorel-35.